frontend/libs/studio-content-library/src/utils/router/QueryParamsRouter.ts

```diff
@@ -0,0 +1,33 @@
+import type { Page } from '../../types';
+
+interface QueryParamsRouter {
+  getCurrentRoute: () => string;
+  navigate: (queryParam: string) => void;
+}
+
+export class QueryParamsRouterImpl implements QueryParamsRouter {
+  private static instance: QueryParamsRouterImpl;
+
+  private constructor() {}
+
+  public static getInstance(): QueryParamsRouterImpl {
+    if (!QueryParamsRouterImpl.instance) {
+      QueryParamsRouterImpl.instance = new QueryParamsRouterImpl();
+    }
+
+    return QueryParamsRouterImpl.instance;
+  }
+
+  public getCurrentRoute(): Page {
+    const searchParams = new URLSearchParams(window.location.search);
+    return searchParams.get('currentLibraryRoute') as string as Page;
```

Comment:
Har du vurdert å plassere `'currentLibraryRoute'` i en variabel?

Reply:
Jeg gjorde det først, men web-storm sier: "Local variable currentRoute is redundant", så jeg valgte å følge det tipset og stole på at metodenavnet var beskrivende nok.

Reply:
Pussig. Den brukes jo flere steder, både her og i `navigate`. Definerte du den lokalt i funksjonen? Den meldingen er noe som typisk kommer opp når noe bare er i bruk ett sted.

Reply:
Kan være jeg missforstod hvor du ville ha denne variabelen. Jeg definerte den lokalt inne i metoden ja, da brukes den bare et sted.

Reply:
Det var ikke mulig å flytte det helt ut. Da jeg flyttet det til en klassevariabel, fikk jeg ikke oppdaterte verdier for searchParams, noe som resulterte i at testene feilet. Jeg gjorde imidlertid en liten forbedring ved å lage en variabel for nøkkelen som brukes til å hente sider. Samtidig endret jeg fra funksjonskall for å hente nåværende side til å bruke en getter.

Reply:
Det synes jeg er mer enn bra nok!